yeat/Snakefile

```diff
@@ -95,18 +95,41 @@ rule downsample:
 
 rule spades:
     output:
-        contigs="analysis/spades/{sample}/contigs.fasta",
-        scaffolds="analysis/spades/{sample}/scaffolds.fasta",
+        contigs="analysis/spades/{sample}_contigs.fasta"
+    input:
+        read1="seq/downsample/{sample}.R1.fq.gz",
+        read2="seq/downsample/{sample}.R2.fq.gz"
+    threads: 8
+    shell:
+        """
+        spades.py -1 {input.read1} -2 {input.read2} -t {threads} -o analysis/spades
+        ln -s scaffolds.fasta {output.contigs}
+        """
+
+
+rule megahit:
+    output:
+        contigs="analysis/megahit/{sample}_contigs.fasta"
     input:
         read1="seq/downsample/{sample}.R1.fq.gz",
         read2="seq/downsample/{sample}.R2.fq.gz",
     threads: 8
-    shell: "spades.py -1 {input.read1} -2 {input.read2} -o analysis/spades/{wildcards.sample}"
+    shell:
+        """
+        megahit -1 {input.read1} -2 {input.read2} -t {threads} -o analysis/megahit-temp
+        mv analysis/megahit-temp/* analysis/megahit
+        rm -r analysis/megahit-temp
```

Comment:
MEGAHIT has a weird quirk where it will not run if the output directory already exists. Because of this, when running MEGAHIT, the output is redirected to the `megahit-temp` directory where all of its' contents will be copied over to the `megahit` directory. After all the files are copied over, the `megahit-temp` directory will be deleted.